src/Interpreters/ProcessList.h

```diff
@@ -223,7 +228,9 @@ class QueryStatus : public WithContext
 
     QueryStatusInfo getInfo(bool get_thread_list = false, bool get_profile_events = false, bool get_settings = false) const;
 
-    CancellationCode cancelQuery(bool kill);
+    /// Cancels the current query.
+    /// Optional argument `exception` allows to set an exception which checkTimeLimit() will throw instead of "QUERY_WAS_CANCELLED".
+    CancellationCode cancelQuery(bool kill, std::exception_ptr exception = nullptr);
```

Comment:
This change was made to keep the same error through the cluster when a backup fails.
If an error happens on a host then we want to cancel the operation on other hosts too - but it seems it's kinder (for anyone who will look at the logs) to keep the same error instead of introducing any other error codes.